src/Yaapii.Xml/XMLSlice.cs

```diff
@@ -1,189 +1,277 @@
-        /// <param name="context"> context information about namespaces in the xml </param>
-        /// <param name="leaf"> is it a document or a node </param>
-        public XMLSlice(XNode node, IXmlNamespaceResolver context) : this(
-            new ScalarOf<XNode>(node),
-            new ScalarOf<IXmlNamespaceResolver>(context)
-        )
-        { }
-
-        public XMLSlice(IScalar<XNode> node, IScalar<IXmlNamespaceResolver> context)
-        {
-            this.cursor = new XMLCursor(node, context);
-        }
-
-        public XNode AsNode()
-        {
-            return this.cursor.AsNode();
-        }
-
-        public IList<IXML> Nodes(string xpath)
-        {
-            return
-                new Atoms.List.Mapped<IXML, IXML>(
-                    node => new XMLSlice(node.AsNode()),
-                    this.cursor.Nodes(xpath)
-                );
-        }
-
-        public IList<string> Values(string xpath)
-        {
-            return this.cursor.Values(xpath);
-        }
-
-        public IXML WithNamespace(string prefix, object uri)
-        {
-            return new XMLSlice(this.cursor.WithNamespace(prefix, uri).AsNode());
-        }
-    }
-}
+﻿// MIT License
+//
+// Copyright(c) 2022 ICARUS Consulting GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using Yaapii.Atoms;
+using Yaapii.Atoms.IO;
+using Yaapii.Atoms.Scalar;
+using Yaapii.Atoms.Text;
+using Yaapii.Xambly;
+
+namespace Yaapii.Xml
+{
+    /// <summary>
+    /// A XML which will slice when you call Nodes(xpath).
+    /// </summary>
+    public sealed class XMLSlice : IXML
+    {
+        private readonly XMLCursor cursor;
+        private readonly IScalar<IXmlNamespaceResolver> context;
+
+        //// <summary>
+        /// A XML which will slice when you call Nodes(xpath).

```

Comment:
i don't understand the description.
what will the XML slice when i call Nodes?
maybe you mean "An Xml that will be sliced when you call Nodes"?
(Eine Xml, die geschnitten/getrennt wird, wenn man Nodes aufruft.)

Reply:
The base type is "IXML" so the comment refers to that. It means that you get an xml object which will execute the slice operation once its method is invoked.